File: backend/src/RepoFluent.Application/Course.cs
namespace RepoFluent.Application;

public sealed record Course(
    string Id,
    string Title,
    string Description,
    string Difficulty,
    int EstimatedMinutes,
    IReadOnlyList<string>? Audience,
    IReadOnlyList<string>? Tags,
    bool IsRequired,
    int Order,
    IReadOnlyList<string>? Prerequisites,
    IReadOnlyList<string>? CompletionRules,
    IReadOnlyList<LearningObjective> Objectives,
    IReadOnlyList<CourseModule> Modules);

File: frontend/projects/api/src/lib/course.ts
import type { CourseModule } from './course-module';
import type { LearningObjective } from './learning-objective';

export interface Course {
  id: string;
  title: string;
  description: string;
  difficulty: string;
  estimatedMinutes: number;
  audience: string[];
  tags: string[];
  isRequired: boolean;
  order: number;
  prerequisites: string[];
  completionRules: string[];
  objectives: LearningObjective[];
  modules: CourseModule[];
}
